x/ibc-transfer/keeper/relay.go

```diff
@@ -70,17 +71,41 @@ func (k Keeper) SendTransfer(
 
 	// begin createOutgoingPacket logic
 	// See spec for this logic: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
-
 	channelCap, ok := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(sourcePort, sourceChannel))
 	if !ok {
 		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
 	}
 
+	// NOTE: denomination and hex hash correctness checked during msg.ValidateBasic
+	prefixedDenom := token.Denom
+
+	// deconstruct the token denomination into the denomination trace info
+	// to determine if the sender is the source chain
+	if strings.HasPrefix(token.Denom, "ibc/") {
+		hexHash := token.Denom[4:]
+		hash, err := types.ParseHexHash(hexHash)
+		if err != nil {
+			return err
+		}
+
+		denomTrace, found := k.GetDenomTrace(ctx, hash)
+		if !found {
+			return sdkerrors.Wrap(types.ErrTraceNotFound, hexHash)
+		}
+		prefixedDenom = denomTrace.GetPrefix() + denomTrace.BaseDenom
```

Comment:
Why are we reconstructing `prefixedDenom` here, shouldn't it be the same as above?